Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -169,5 +162,50 @@ - (nonnull NSArray *)filterResponseInterceptors:(nonnull NSArray *)proposedRespo
     return [NSArray arrayWithArray:responseInterceptors];
 }
 
+- (void)processResponse:(NSURLResponse *)response onThread:(NSThread *)thread
+{
+    __block CDTHTTPInterceptorContext *ctx =
+    [[CDTHTTPInterceptorContext alloc] initWithRequest:[self.request mutableCopy]];
+
+    ctx.response = (NSHTTPURLResponse*)response;
+    for (NSObject<CDTHTTPInterceptor> *obj in self.responseInterceptors) {
+        ctx = [obj interceptResponseInContext:ctx];
+    }
+
+    if (ctx.shouldRetry && self.remainingRetries > 0) {
+        // retry
+        self.remainingRetries--;
+        self.inProgressTask = [self makeRequest];
+        [self.inProgressTask resume];
+    } else {
+        MYOnThread(thread, ^{
+            [self.delegate receivedResponse:response];
+        });
+        self.finished = YES;
+    }
+}
+
+- (void)processError:(NSError *)error onThread:(NSThread *)thread
```

Comment:
does `processError` change the handling of response interceptors in case of errors?

I guess previously the only way that the interceptors could handle errors would be to introspect the value of the HTTP status code in the NSHTTPURLResponse object. 

Maybe contexts need to have an extra `error` field, although this is could be considered a breaking change.